dashboard/app/controllers/pd/teachercon1819_registration_controller.rb

```diff
@@ -66,13 +66,14 @@ def partner
       return
     end
 
-    unless current_user.regional_partners.where(group: 3).any?
+    regional_partner = current_user.regional_partners.find_by(group: 3)
+    unless regional_partner
       render :only_group_3
       return
     end
 
-    regional_partner = current_user.regional_partners.first
-    teachercon = get_matching_teachercon(regional_partner)
+    teachercon = Pd::Application::RegionalPartnerTeacherconMapping::TEACHERCONS.detect {|tc| tc[:city] == params[:city].titleize}
```

Comment:
`Pd::Application::RegionalPartnerTeacherconMapping` is already [included](https://github.com/code-dot-org/code-dot-org/blob/c97019e0510f8ee0e74295fcd85f414184c11324/dashboard/app/controllers/pd/teachercon1819_registration_controller.rb#L2) in this class, so there's no need to reference it explicitly.

Do we no longer care if the regional partner is associated with a teachercon? Previously `get_matching_teachercon` ensured that. I'd suggest adding an optional `city:` param to that method, and calling that here. If instead we don't care if the partner matches, then inspecting `TEACHERCONS` is fine.

Reply:
We still need to store this `teachercon` data in the registration form somewhere, so it can be later queried and displayed.

Reply:
I'll follow up with Tanya on the first part.